File: psc-backend/apps/inspection/audit/serializers/checklist.py
"""Serializers for Audit checklist master reads."""

from __future__ import annotations

from rest_framework import serializers

from apps.inspection.audit.services.checklist import AuditChecklistBundle


class AuditChecklistResponseSerializer(serializers.Serializer):
    def to_representation(self, instance: AuditChecklistBundle):
        checklist = instance.checklist
        return {
            "audit_id": str(instance.audit_detail.id),
            "selected": checklist is not None,
            "ship_type_filter": instance.ship_type_filter,
            "item_filter_applied": instance.item_filter_applied,
            "checklist": None
            if checklist is None
            else {
                "id": str(checklist.id),
                "checklist_code": checklist.checklist_code,
                "name": checklist.name,
                "auditee_type": checklist.auditee_type,
                "scope_dept": checklist.scope_dept,
                "ship_type_scope": checklist.ship_type_scope,
                "source_form_ref": checklist.source_form_ref,
                "code_version": checklist.code_version,
            },
            "items": [
                {
                    "id": str(item.id),
                    "location_code": item.location_code or "",
                    "item_code": item.item_code,
                    "question": item.question,
                    "guideline": item.guideline or "",
                    "regulation_ref": item.regulation_ref or "",
                    "ksm_sms_ref": item.ksm_sms_ref or "",
                    "ship_type": item.ship_type or "",
                    "sequence_no": item.sequence_no,
                }
                for item in instance.items
            ],
        }

File: psc-frontend/src/schemas/audit/checklist.ts
export type AuditChecklistWalkStatus = 'NOT_REVIEWED' | 'COMPLIANT' | 'ADD_FINDING';

export interface AuditChecklistItem {
  id: string;
  location_code: string;
  item_code: string;
  question: string;
  guideline: string;
  regulation_ref: string;
  ksm_sms_ref: string;
  ship_type: string;
  sequence_no: number;
}

export interface AuditChecklistHeader {
  id: string;
  checklist_code: string;
  name: string;
  auditee_type: string;
  scope_dept: string | null;
  ship_type_scope: string | null;
  source_form_ref: string;
  code_version: string | null;
}

export interface AuditChecklist {
  audit_id: string;
  selected: boolean;
  ship_type_filter: string | null;
  item_filter_applied: boolean;
  checklist: AuditChecklistHeader | null;
  items: AuditChecklistItem[];
}
